doc/source/dev/development_gitpod.rst

```diff
@@ -265,6 +265,35 @@ GitHub application page.
 .. image:: ./gitpod-imgs/gitpod-edit-permissions-repo.png
    :alt: Gitpod integrations - edit GH repository permissions screenshot
 
+
+Generic debugging/testing steps for gitpod environment. 
+******************************************************************************************
+
+A user might use the below steps one after another to debug/test Gitpod 
+environment for its correctness:
+
+#. Make sure that the Pre-Build is completed successfully without any errors. 
+   By referring to the commands used to build a gitpod container in 
+   `gitpod.yml <https://github.com/numpy/numpy/blob/main/.gitpod.yml>`_, it is 
+   possible to pinpoint which command caused the build to fail. For example -
+   
+   .. image:: ./gitpod-imgs/example_error_log.png
+      :alt: Gitpod integrations - Example error log
+
+   
+   In the above log file, when compared to ``gitpod.yml``. Observe that the 
+   error message came due to the command run after the log “Building Docs”. 
+   Hence “make html” is the command that failed. Knowing which command caused 
+   the build to break failed is the first step to debugging.
+#. Verify that the dependencies installed match the ones listed in 
+   ``environment.yml``. Use command``conda list`` to list all packages 
+   and their versions.
+#. Verify if it is not one of the installed dependencies causing the build to break. 
+   Dependencies causing build breakage are more prevalent in those whose versions 
+   are not pinned with ``==`` in ``environment.yml`` as no version specified means 
```

Comment:
I think you meant `=`, on a side note, why do we not pin all the packages like in `*_requirements.txt` and not bump till depandabot complains, it makes sense from a security perspective as well?

Reply:
Yes, You are correct. I meant `=`. Not sure about the side note tho. `environment.yml` has been like this since I started contributing and your argument makes sense to be added. 
I think @seberg @rossbar could add more valuable insights to this discussion.